std/utf.d

```diff
@@ -1905,6 +1908,14 @@ unittest
 }
 /* =================== Encode ======================= */
 
+private dchar _utfException(UseReplacementDchar useReplacementDchar)(string msg, dchar c)
```

Comment:
Unfortunately I think this needs to be rediscussed. Abusing replacementDchar which is intended for GUIs to replace our UTFExceptions is a mistake IMO. It also doesn't eliminate redundant validation.
The scheme that works much better IMO is to load raw data as ubyte[], validate it to become char[] and then assume char[] is valid.